lib/matplotlib/scale.py

```diff
@@ -193,9 +192,9 @@ def set_default_locators_and_formatters(self, axis):
         # update the minor locator for x and y axis based on rcParams
         if (axis.axis_name == 'x' and mpl.rcParams['xtick.minor.visible'] or
                 axis.axis_name == 'y' and mpl.rcParams['ytick.minor.visible']):
-            axis.set_minor_locator(AutoMinorLocator())
+            axis.minorticks_on()
```

Comment:
This line still isn't being hit by the tests?   Can we add a test here?

Reply:
I think the `test_minorticks_rc` in `test_ticker.py` only covers `linear` scale. I can add a test.